data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -133,12 +134,13 @@ BulkResponse getResponse() {
         }
     }
 
-    public BulkRetryStrategy(final RequestFunction<AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest>, BulkResponse> requestFunction,
+    public BulkRetryStrategy(final OpenSearchSink sink, final RequestFunction<AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest>, BulkResponse> requestFunction,
```

Comment:
Rather than provide the `OpenSearchSink`, make an interface for this.

```
interface ResponseLatencyAware {
   void updateResponseLatency(Duration latency);
}
```

If you do the refactoring I suggest for the `BulkRequestExecutor`, then that class can implement this. Otherwise, `OpenSearchSink` would implement it.

Regardless, we should not be coupling the code that is here.